File: python/quote_items.py
import sdk.python.entities
from sdk.python.util.item_types import item_type
from sdk.python.entities import Property


class QuoteItem(sdk.python.entities.Entity):

    resource = '/quote_items'
    json_name = 'quote_item'

    id = Property(int)
    type = Property(int)
    quantity = Property(int)
    description = Property(str)
    unit_price = Property(float)  # unit cents in db

    tax_type = Property(sdk.python.country_taxes.CountryTax)

    def item_total(self):
        """ Calculate the total of the item by
            multiplying the unit_price and quantity. It then
            returns the total
        """
        return self.quantity * self.unit_price

    def item_type_name(self):
        """ Return name of the quote_item type instead of type id """
        return item_type[self.type]


class QuoteItems(sdk.python.entities.Resource):

    entity_class = QuoteItem
    json_name = 'quoteItem'


quote_items = QuoteItems()

File: php/src/quote_items.php
<?php
declare(strict_types=1);

require_once 'entity.php';
require_once 'util/item_types.php';

class QuoteItem extends Entity
{
    public static $resource = '/quote_items/';
    public static $json_name = 'quote_item';


    public function __construct()
    {
        $this->json_property('id', 'integer');
        $this->json_property('type', 'integer');
        $this->json_property('quantity', 'integer');
        $this->json_property('description', 'string');
        $this->json_property('unit_price', 'float');
    }

    public function item_total()
    {
        /*
            Calculate the total of the item by
            multiplying the unit_price and quantity. It then
            returns the total
        */

        $total = $this->quantity * $this->unit_price;
        return $total;
    }

    public function item_type_name()
    {
        // return name of the quote item type instead of type id
        return ITEM_TYPE[$this->type];
    }
}

class QuoteItems extends resource
{
    public function __construct()
    {
        $this->entity_class = 'QuoteItem';
        $this->json_name = 'quoteItem';
    }
}

$quote_items = new QuoteItems();
